tests/dictzip_test.py

```diff
@@ -17,33 +20,53 @@
 """
 
 
-# TODO Test also dictzip func
-# TODO Check if dictzip in GH Action and avoid if not
-class AsciiLowerUpperTest(TestGlossaryErrorsBase):
-	def make_dz(self, path: Path) -> Path:
+class TestDictzipBase(TestGlossaryErrorsBase):
+	def _run(self, func: FUNC_TYPE, filename: str | Path) -> None:
+		done = func(str(filename))
+		if not done:
+			self.skipTest(f'Missing dependency for {func.__name__}')
+
+
+class TestDictzip(TestDictzipBase):
+	def make_dz(self, func: FUNC_TYPE, path: str | Path) -> Path:
 		"""Get path of dzipped file contains TEXT."""
 		test_file_path = Path(path)/"test_file.txt"
 		result_file_path = test_file_path.parent/(test_file_path.name + ".dz")
 		with open(test_file_path, "a") as tmp_file:
 			tmp_file.write(TEXT)
-		runDictzip(str(test_file_path))
+		self._run(func, test_file_path)
 		return result_file_path
 
-	def test_compressed_exists(self):
+	def is_compressed_exists(self, func: FUNC_TYPE) -> None:
 		with tempfile.TemporaryDirectory() as tmp_dir:
-			result_file_path = self.make_dz(tmp_dir)
+			result_file_path = self.make_dz(func, tmp_dir)
 			self.assertTrue(result_file_path.exists())
 			self.assertTrue(result_file_path.is_file())
 
-	def test_compressed_matches(self):
+	def is_compressed_matches(self, func: FUNC_TYPE) -> None:
 		with tempfile.TemporaryDirectory() as tmp_dir:
-			result_file_path = self.make_dz(tmp_dir)
+			result_file_path = self.make_dz(func, tmp_dir)
 			with gzip.open(result_file_path, 'r') as file:
 				result = file.read().decode()
 		self.assertEqual(result, TEXT)
 
-	def test_missing_target(self):
-		filename = "/NOT_EXISTING_PATH/file.txt"
-		expected_msg = f"[Errno 2] No such file or directory: '{filename}'"
-		runDictzip(filename)
-		self.assertLogError(expected_msg)
+	test_idzip_compressed_exists = partialmethod(is_compressed_exists, _idzip)
+	test_idzip_compressed_matches = partialmethod(is_compressed_matches, _idzip)
+
+	test_dictzip_compressed_exists = partialmethod(is_compressed_exists, _dictzip)
+	test_dictzip_compressed_matches = partialmethod(is_compressed_matches, _dictzip)
+
+
+class TestDictzipErrors(TestDictzipBase):
+	def tearDown(self):
+		self.mockLog.clear()
+		super().tearDown()
+
+	def on_missing_target(self, func: FUNC_TYPE) -> None:
+		filename = '/NOT_EXISTED_PATH/file.txt'
+		self._run(func, filename)
+		err_num = self.mockLog.printRemainingErrors()
+		self.assertEqual(err_num, 1)
```

Comment:
Not a good test. Test the error message, or don't test the error it at all.

Reply:
Since error message contains external output, which is not controlled by PyGlossary, it is not reliable to check an exact string. Nevertheless I would suggest to not to delete the case: e. g. it helped me to find misbehavior in old `dictzip` errors handling.

There is the following visible options:
- Add a method to match a substring of an error message.
- Raise and catch an exception (which is not a way of the PyGlossary as seems). 
- Write an exact error message and then write a command output with e. g. `info` severity (then test it in a usual PyGlossary way).

Reply:
Substring is fine (long enough to be unique).